src/sentry/lang/native/minidump.py

```diff
@@ -201,8 +215,27 @@ def get_attached_minidump(data):
 def merge_symbolicator_minidump_response(data, response):
     sdk_info = get_sdk_from_event(data)
 
-    # TODO(markus): Add OS context here when `merge_process_state_event` is no
-    # longer called for symbolicator projects
+    data['platform'] = 'native'
+    data['level'] = 'fatal' if response['crashed'] else 'info'
+
+    if response.get('timestamp'):
+        data['timestamp'] = float(response['timestamp'])
+
+    if response.get('system_info'):
+        info = response['system_info']
+        set_path(data, 'contexts', 'os', 'type', value='os')  # Required by "get_sdk_from_event"
+        if info.get('os_name'):
+            set_path(data, 'contexts', 'os', 'name',
+                     value=MINIDUMP_OS_TYPES.get(info['os_name'], info['os_name']))
+
+        if info.get('os_version'):
```

Comment:
Also, here please use something closer to what `wip/symbolicator` has. Directly copied (`self.setdefault` was a wrapper around `setdefault_path`):

```py
    def _apply_system_info(self, system_info):
        self.setdefault(['contexts', 'os', 'name'], system_info.get('os_name'))
        self.setdefault(['contexts', 'os', 'version'],
                        system_info.get('os_version'))
        self.setdefault(['contexts', 'os', 'build'],
                        system_info.get('os_build'))
        self.setdefault(['contexts', 'device', 'arch'],
                        system_info.get('cpu_arch'))
```

Reply:
done